File: my-solutions/0011-container-with-most-water/solution.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        maxWater, l, r = float('-inf'), 0, len(height) - 1
        while(l < r):
            maxWater = max(maxWater, min(height[l], height[r]) * (r - l))
            if height[l] > height[r]:
                r -= 1
            else:
                l += 1
        
        return maxWater

File: my-solutions/0011-container-with-most-water/solution.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int maxArea = 0;
        int currArea = 0;

        int l = 0, r = size(height) - 1;
        while(l < r) {
            currArea = min(height[l], height[r]) * (r - l);
            if(height[l] < height[r]) ++l;
            else --r;
            maxArea = max(maxArea, currArea);
        }
        return maxArea;
    }
};
